lib/Cake/Test/Case/Model/ModelReadTest.php

```diff
@@ -8531,4 +8531,48 @@ public function testRelatedAfterFindCallback() {
 		);
 		$this->assertEquals($expected, $results, 'Model related with belongsTo afterFind callback fails');
 	}
+
+	public function testQueryRespectsCacheQueriesParam() {
+		$model = new User();
+		$model->save(array('user' => 'Chuck'));
+		$userTableName = $this->db->fullTableName('users');
+
+		$getUserNameFromDb = function ($withParams = true, $cacheArgument = null) use ($model, $userTableName) {
+			if ($withParams) {
+				$query = sprintf('SELECT u.user FROM %s u WHERE id=?', $userTableName);
+
+				if ($cacheArgument !== null) {
+					$users = $model->query($query, array($model->id), $cacheArgument);
+				} else {
+					$users = $model->query($query, array($model->id));
+				}
+			} else {
+				$query = sprintf('SELECT u.user FROM %s u ORDER BY id DESC LIMIT 1', $userTableName);
+
+				if ($cacheArgument !== null) {
+					$users = $model->query($query, $cacheArgument);
+				} else {
+					$users = $model->query($query);
+				}
+			}
+
+			return isset($users[0][0]) ? $users[0][0]['user'] : $users[0]['u']['user'];
```

Comment:
This is a pretty complex piece of test code. Could we simplify it by breaking down each of the cases into separate test methods? I find test code is easier to understand when the logic is simple, and there is some duplication.